### PR TITLE
[#3868] Improvement(spark-connector): Invoking `invalidateTable` to remove Iceberg catalog cache

### PR DESCRIPTION
**Pull request by caican00**
### What changes were proposed in this pull request?

Invoking `invalidateTable` to remove cache after doing some table ddl operations, such as `dropTable`, `renameTable`, `alterTable` and `purgeTable`.

### Why are the changes needed?
To avoid unexpected exceptions that the table cache cannot be cleaned when the config `cache-enabled` of iceberg catalog is set to `true`.

Fix: https://github.com/datastrato/gravitino/issues/3868

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?
Changing the `cache-enabled` to `true` in `SparkEnvIT`  and test based on an existing test case, `testDropAndWriteTable`

### CODE REVIEW

**Comment by caican00** [comment]
@FANNG1 could you help review this pr? Thank you.

**Comment by FANNG1** [comment]
> @FANNG1 could you help review this pr? Thank you.

just one comment

File: spark-connector/spark-common/src/test/java/com/datastrato/gravitino/spark/connector/integration/test/SparkEnvIT.java [hunk 1]
```diff
@@ -171,17 +175,25 @@ private void initHdfsFileSystem() {
...
+            .set("spark.plugins", GravitinoSparkPlugin.class.getName())
+            .set(GravitinoSparkConfig.GRAVITINO_URI, gravitinoUri)
+            .set(GravitinoSparkConfig.GRAVITINO_METALAKE, metalakeName)
+            .set(GravitinoSparkConfig.GRAVITINO_ENABLE_ICEBERG_SUPPORT, "true")
+            .set("hive.exec.dynamic.partition.mode", "nonstrict")
+            .set("spark.sql.warehouse.dir", warehouse)
+            .set("spark.sql.session.timeZone", TIME_ZONE_UTC);
+    if (lakeHouseIcebergProvider.equalsIgnoreCase(getProvider())) {
+      sparkConf.set(
+          String.format("spark.sql.catalog.%s.%s", getCatalogName(), ICEBERG_CATALOG_CACHE_ENABLED),
```
[the remark this answers is not in the document]
Comment (caican00) [reply spark-connector/spark-common/src/test/java/com/datastrato/gravitino/spark/connector/integration/test/SparkEnvIT.java:189]: ok

[the remark this answers is not in the document]
Comment (caican00) [reply spark-connector/spark-common/src/test/java/com/datastrato/gravitino/spark/connector/integration/test/SparkEnvIT.java]: updated.

**Comment by caican00** [comment]
comments have been addressed. cc @FANNG1

**Review by FANNG1 — APPROVED** [review]

**Pull request merged**